jetbrains-core/src/software/aws/toolkits/jetbrains/utils/SpinUtils.kt

```diff
@@ -20,3 +20,18 @@ fun spinUntil(duration: Duration, condition: () -> Boolean) {
         }
     }
 }
+
+/**
+ * Keeps running the function until it returns a non-null value. Checks every 100ms
+ */
+suspend fun <T> spinUntilResult(duration: Duration, func: () -> T?): T {
```

Comment:
looks familiar, does this appear elsewhere, or am i thinking of another codebase?

Reply:
Yaaaa, its similar to software/aws/toolkits/core/utils/Waiter.kt:51